node/resolvers/auth/index.ts

```diff
@@ -15,47 +15,59 @@ const makeRequest = async (ctx, url) => {
   return await http.request(await configRequest(ctx, url))
 }
 
+const getSessionToken = async ioContext => {
+  const { data, status } = await makeRequest(ioContext, paths.getTemporaryToken(ioContext.account, ioContext.account))
+  if (!data.authenticationToken) {
+    throw new ResolverError(`ERROR ${data}`, status)
+  }
+  return data.authenticationToken
+}
+
+const setVtexIdAuthCookie = (headers, ioContext, response) => {
+  const authAccount = `VtexIdclientAutCookie_${ioContext.account}`
+  const authCookie = parse(headers['set-cookie'].find(checkAuth => {
+    return checkAuth.includes(authAccount)
+  }))
+  response.set('Set-Cookie', serialize(authAccount, authCookie[authAccount], {
+    httpOnly: true,
+    path: '/',
+    maxAge: new Date(authCookie['expires']).getTime(),
+    secure: true
+  }))
+}
+
 export const mutations = {
   sendEmailVerification: async (_, args, { vtex: ioContext, response }) => {
-    const { data, status } = await makeRequest(ioContext, paths.getTemporaryToken(ioContext.account, ioContext.account))
-    if (!data.authenticationToken) {
-      throw new ResolverError(`ERROR ${data}`, status)
-    }
-    await makeRequest(ioContext, paths.sendEmailVerification(args.email, data.authenticationToken))
-    response.set('Set-Cookie', serialize('VtexTemporarySession', data.authenticationToken, { httpOnly: true, secure: true, path: '/' }))
+    const VtexSessionToken = await getSessionToken(ioContext)
+    await makeRequest(ioContext, paths.sendEmailVerification(args.email, VtexSessionToken))
+    response.set('Set-Cookie', serialize('VtexSessionToken', VtexSessionToken, { httpOnly: true, secure: true, path: '/' }))
```

Comment:
This cookie should last up to 9 min, because the token you're setting as the session only lasts for 10min.